Setup python package and Docker image publishing

pracucci opened the pull request:
In this PR I've done the setup to publish the python package and Docker image. I have not tested it yet, cause I would like to get a quick feedback on the overall changes. If they look good, I will publish a first version to test it.

Commit:
Setup python package and Docker image publishing

Commit:
Updated README

esanchezm commented on README.md:
````diff
@@ -48,6 +54,33 @@ python3 -m unittest
 ```
 
 
+### How to publish a new version
````
I think this is more internal doc than something anyone would need on a README. If you really want it, move it in a Wiki page of the github repo.

pracucci replied:
I would prefer to be able to manage it through the code editor. What do you think about `PUBLISH.md`?

esanchezm replied:
Sure, is way better :)

esanchezm commented on README.md:
```diff
@@ -21,6 +21,12 @@ Safety countermeasures:
+
+1. Manually install and run the [`aws-cloud-unmap` Python package](https://pypi.org/project/aws-cloud-unmap/)
+2. Use the [Docker image available on Docker hub](https://hub.docker.com/u/spreaker/aws-cloud-unmap/)
+
```
I think it's easier to include the exact commands to install and run for the first time. That's something I really appreciate when reading a README

pracucci replied:
How does it look, now?

esanchezm replied:
Way better! Just after the `pip3`, you should add a running example similar to the docker one.

esanchezm commented on setup.py:
```diff
@@ -0,0 +1,30 @@
+  keywords         = ['aws', 'cloud map'],
+  classifiers      = [],
+  python_requires  = ' >= 3',
+  install_requires = ['boto3==1.9.123', 'python-json-logger==0.1.10'],
```
You could put the content of `requirements.txt`  here so you only mantain one file.

pracucci replied:
What's about dev/test dependencies in `requirements.txt`? May you give me an example, please?

esanchezm replied:
Normally you would have a `requirements.txt` per environment. But one thing you could is keep this as is for releases and have another `requirements-dev.txt` like this:

```
-r requirements.xt
dev_req==1.0.0
```

Commit:
Moved publish instructions to a dedicated file

Commit:
Added run example to README

Commit:
Moved dev requirements to requirements-dev.txt

Commit:
Load requirements.txt in setup.py

Commit:
Moved version in a varible in setup.py

esanchezm commented on setup.py:
```diff
@@ -21,7 +25,7 @@
   classifiers      = [],
   python_requires  = ' >= 3',
-  install_requires = ['boto3==1.9.123', 'python-json-logger==0.1.10'],
+  install_requires = [requirements],
```
`requirements` is already a list

esanchezm replied:
I'll approve because it's trivial

Commit:
Added example cli when manually installed

Commit:
Fixed install_requires in setup.py

esanchezm commented:
Nice work! 👍

Commit:
Updated .gitignore

Commit:
Fixed deps installation in Dockerfile-dev

Commit:
Fixed package description content type in setup.py

Commit:
Fixed requirements and entrypoint in setup.py

pracucci commented:
@esanchezm I did some changes since your approval and I would like a re-review:
1. Upload package with `twine`
2. Moved dependencies to `setup.py` and load them from `setup.py` in `requirements.txt` (otherwise we need to add `requirements.txt` to the package in order for the installation to work correctly)
3. Added the function `run()` which is the one referenced in `entry_points` in `setup.py`

esanchezm commented:
I see you find a different approach for the `requirements.txt`. Great!

Commit:
Added AWS credentials to Docker cli example
